File: IT/Oggetti/Esercizi/C++/esercizio23.cpp
/* file esercizio23.cpp */

#include <iostream>

using std::cin;
using std::cout;
using std::endl;
using std::ostream;

class Point2d {
	private:
		double x;
		double y;

	public:
		Point2d(double, double);
		double getX();
		void setX(double);
		double getY();
		void setY(double);

	friend ostream& operator<<(ostream& os, const Point2d&);
};

Point2d::Point2d(double x, double y) {
	this->setX(x);
	this->setY(y);
}

double Point2d::getX() {
	return this->x;
}

void Point2d::setX(double x) {
	this->x = x;
}

double Point2d::getY() {
	return this->y;
}

void Point2d::setY(double y) {
	this->y = y;
}

ostream& operator<<(ostream& os, const Point2d& p) {
	return os << "(" << p.x << ", " << p.y << ")";
}

int main() {
	Point2d o(0.0, 0.0);
	Point2d p(1.0, 2.0);

	cout << "O = " << o << " P = " << p << endl;
}

File: IT/Oggetti/Esercizi/Java/Esercizio23.java
/* file Esercizio23.java */

class Point2d {
	private double x;
	private double y;

	public Point2d(double x, double y) {
		this.setX(x);
		this.setY(y);
	}

	public double getX() {
		return this.x;
	}

	public void setX(double x) {
		this.x = x;
	}

	public double getY() {
		return this.y;
	}

	public void setY(double y) {
		this.y = y;
	}

	@Override
	public String toString() {
		return "(" + this.x + ", " + this.y + ")";
	}
}

public class Esercizio23 {
	public static void main(String [] args) {
		Point2d o = new Point2d(0, 0);
		Point2d p = new Point2d(1, 2);

		System.out.println("O = " + o + " P = " + p);
	}
}

